### DISCUSSION TITLE
Can an additional description be added to the marked flow to differentiate between different types of marks？

### DISCUSSION THREAD

**Discussion by xzhacker123** [mitmproxy/mitmproxy#6843] [General]
I will filter flows based on certain conditions and mark them accordingly. Typically, I print information in the Events log. Is there a way to differentiate between different types of marks in the flowlist?
e.g.
```
def response(self, flow: http.HTTPFlow):
    if 'X-Powered-by' in flow.response.headers:
        flow.marked = True
        logging.error(f"X-Powered-by:{flow.response.headers['X-Powered-by']}")
```

**Comment by Prinzhorn** [comment]
`marked` can be an emoji name, e.g. `flow.marked = ':grapes:'`

https://docs.mitmproxy.org/stable/api/mitmproxy/flow.html#Flow.marked

You can also add additional data via `flow.comment`.

**Reply by xzhacker123** [reply]
Thank you for the method you provided, it is effective. However, I previously used the `examples/contrib/all_markers.py` addon and it didn't display emojis.

**Reply by Prinzhorn** [reply]
Thanks, I'll fix the example
